pkg/admission/workspacetypeexists/admission.go

```diff
@@ -132,8 +137,8 @@ func (o *workspacetypeExists) Admit(ctx context.Context, a admission.Attributes,
 	}
 
 	// if the user has not provided any type, use the default from the parent workspace
-	empty := tenancyv1alpha1.WorkspaceTypeReference{}
-	if ws.Spec.Type == empty {
+	// We default to workspaceType only if not mount.
+	if ws.Spec.Type == nil || *ws.Spec.Type == (tenancyv1alpha1.WorkspaceTypeReference{}) {
```

Comment:
This feels a little odd as a comparison, shouldn't we simply check for `ws.Spec.Type.Name == ""`?

Reply:
I think this is better now: `if ws.Spec.Type == nil || (ws.Spec.Type != nil && ws.Spec.Type.Name == "") {`

Reply:
> There is a semi-breaking change where type now becomes a pointer and optional, so it can be nil now.

Per se yes, but because you are already checking `ws.Spec.Type == nil` in the first part of the condition, the part after `||` will only ever be evaluated if `Type` is not `nil`. So we don't need another check for that.